src/Git.hs

```diff
@@ -353,6 +376,27 @@ runGit userConfig repoDir operation =
       exists <- liftIO $ doesDirectoryExist (repoDir </> ".git")
       pure $ cont exists
 
+    PushAtomic refs cont -> do
+      result <- callGitInRepo $ ["push", "--atomic", "origin"] ++ map refSpec refs
+      let pushResult = case result of
+            Left  _ -> PushRejected
+            Right _ -> PushOk
+      when (pushResult == PushRejected) $ logInfoN "atomic push was rejected"
+      pure $ cont pushResult
+    LastTag sha cont -> do
+      result <- callGitInRepo ["describe", "--abbrev=0", "--tags", refSpec sha]
+      pure $ cont $ either (const Nothing) Just result
```

Comment:
Obtaining only the last tag is problematic in cases where accidentally (or maybe intentionally, for a different purpose) a tag with a different naming convention than the expected one is pushed.

Reply:
Answered below :arrow_double_down:

Reply:
But actually - in this line we only handle the case where there is no previous tag whatsoever e.g. in a fresh repository